### PR TITLE
ref(hc): Remove usages of orgmember.user from tests

### PR DESCRIPTION
**Pull request by corps**
Part of the work of breaking down https://github.com/getsentry/sentry/pull/48798/files

Trying to just get the tests half of this passing and in on master to reduce the diff size of future chunks and increase the ease of review.

### CODE REVIEW

File: tests/sentry/api/endpoints/test_team_members.py [hunk 1]
```diff
@@ -46,19 +47,20 @@ def test_team_members_list_does_not_include_invite_requests(self):
         response = self.get_response(self.org.slug, self.team.slug)
         assert response.status_code == 200
         assert len(response.data) == 2
-        assert response.data[0]["id"] == str(self.team_member.id)
-        assert response.data[1]["id"] == str(pending_invite.id)
+        assert response.data[1]["id"] == str(self.team_member.id)
+        assert response.data[0]["id"] == str(pending_invite.id)
```
Comment (GabeVillalobos) [inline tests/sentry/api/endpoints/test_team_members.py:51]: Did the order on these response entries reverse? If so, is this something we need to update our docs on?

**Review by GabeVillalobos — APPROVED** [review]
Tests are failing, but otherwise looks good to me!

File: src/sentry/testutils/helpers/task_runner.py [hunk 2]
```diff
@@ -9,11 +9,12 @@
...
 @contextmanager
 def TaskRunner():
+    prev = settings.CELERY_ALWAYS_EAGER
     settings.CELERY_ALWAYS_EAGER = True
     current_app.conf.CELERY_ALWAYS_EAGER = True
     yield
-    current_app.conf.CELERY_ALWAYS_EAGER = False
-    settings.CELERY_ALWAYS_EAGER = False
+    current_app.conf.CELERY_ALWAYS_EAGER = prev
+    settings.CELERY_ALWAYS_EAGER = prev
```
Comment (markstory) [inline src/sentry/testutils/helpers/task_runner.py:17]: Did we have nested taskrunner contexts?

**Review by markstory — APPROVED** [review]

File: src/sentry/testutils/helpers/task_runner.py (hunk 2 above)
Comment (corps) [reply src/sentry/testutils/helpers/task_runner.py:17]: Yes, we did, unfortunately.  This fixed subtle bugs as a result of that.

**Pull request merged**